bundles/org.jkiss.utils/src/org/jkiss/utils/BeanUtils.java

```diff
@@ -431,6 +431,20 @@ public static <T> T getFieldValue(@NotNull Object object, @NotNull String name)
         return (T) field.get(object);
     }
 
+    @Nullable
+    public static Object handleObjectMethod(@NotNull Object proxy, @NotNull Method method, Object[] args) {
+        switch (method.getName()) {
+            case "toString":
+                return "Proxy";
+            case "hashCode":
+                return System.identityHashCode(proxy);
+            case "equals":
+                return proxy == args[0];
```

Comment:
Why so strange equals implementation?
Also why do you use custom Object methods instead of calling them from `proxy` object?

Reply:
The `args[0]` stores the argument of the `equals` method, so we compare it with the proxy.

We can't use `equals` on the proxy itself, as it will result in recursion.